src/bsdfs/mask.cpp

```diff
@@ -220,6 +220,12 @@ class MaskBSDF final : public BSDF<Float, Spectrum> {
         return dr::clamp(m_opacity->eval_1(si, active), 0.f, 1.f);
     }
 
+    Spectrum eval_diffuse_reflectance(const SurfaceInteraction3f &si,
+                                      Mask active) const override {
+        Float opacity = eval_opacity(si, active);
+        return opacity * (m_nested_bsdf->eval_diffuse_reflectance(si, active));
```

Comment:
Not sure this is correct. I think the albedo should be independent of the opacity here. E.g. when the opacity is very low, with the current code the albedo would be almost black which seems incorrect.
```suggestion
        return m_nested_bsdf->eval_diffuse_reflectance(si, active);
```